Denisov.Pavel #Lab1

Ppasha9 opened the pull request:
- "first" final version of daemon

Commit:
Add daemon

- "first" final version of daemon

Commit:
Fix pid file mistakes

- create pid file in 'build.sh' script
- do not remove pid file in daemon code

KotsMikhail commented on denisov.pavel/lab1/common.h:
```diff
@@ -0,0 +1,16 @@
+
+#define WARNING(...) do{ syslog(LOG_WARNING, ##__VA_ARGS__); } while(0)
+#define ERROR(...)   do{ syslog(    LOG_ERR, ##__VA_ARGS__); } while(0)
+#define INFO(...)    do{ syslog(   LOG_INFO, ##__VA_ARGS__); } while(0)
```
Тут вроде не нужен do{ } while(0), один же вызов функции

KotsMikhail commented on denisov.pavel/lab1/common.h:
```diff
@@ -0,0 +1,16 @@
+#define ERROR(...)   do{ syslog(    LOG_ERR, ##__VA_ARGS__); } while(0)
+#define INFO(...)    do{ syslog(   LOG_INFO, ##__VA_ARGS__); } while(0)
+
+#define LOG_ERROR_AND_EXIT(...) \
```
Почему не сделать это inline функцией вместо define? Директивы препроцессора для этой цели уже сильно устарели.

Ppasha9 replied:
Согласен
Изначально это было сделано для того, чтобы не писать template variadic функции
В итоге решил просто напрямую вызывать syslog

KotsMikhail reviewed (changes requested):
В целом неплохо, кроме небольших замечаний по коду
Логирование стоит настроить с помощью openlog, чтобы в лог выводился хотя бы pid процесса
Так же демон не должен спамить в лог кучу сообщений каждые 30 секунд, это удобно для отладки, но не очень удобно для пользователя

Commit:
Fix mistakes

- use 'openlog' function
- exchange defines WARNING, INFO and ERROR by directly calling 'syslog' function
- delete unused logging

Ppasha9 commented:
За совсем про openlog(
Добавил

KotsMikhail reviewed (approved):
Засчитано
